js/bundle.go

```diff
@@ -60,7 +60,16 @@ type BundleInstance struct {
 
 // NewBundle creates a new bundle from a source file and a filesystem.
 func NewBundle(src *loader.SourceData, filesystems map[string]afero.Fs, rtOpts lib.RuntimeOptions) (*Bundle, error) {
-	compiler, err := compiler.New()
+	compatMode := compiler.CompatibilityModeES6
+	if rtOpts.CompatibilityMode.Valid {
+		if cm, err := lib.ValidateCompatibilityMode(rtOpts.CompatibilityMode.String); err == nil {
+			compatMode = cm
+		} else {
+			return nil, err
+		}
+	}
+
```

Comment:
I think this and the one below should be done in the configuration parsing and lib.RuntimeOptions should have the `CompatabilityMode` type field. But as we are not doing this for anything else, I am not particularly certain if it's worth the effort, again. Also I am pretty sure it's going to make the configuration even messier

Reply:
You're right, `NewBundle*` should receive validated `RuntimeOptions`, where preferably `CompatibilityMode` is already of type `compiler.CompatibilityMode`, and not `string`. I didn't see a good place to do this validation/conversion, though.

`getRuntimeOptions` seems like the only place, since `newRunner` is called immediately after. Let me know if that would be an improvement, but yeah, this should be fixed in #883.